paddle/fluid/operators/detail/grpc_server.cc

```diff
@@ -157,9 +157,12 @@ class RequestPrefetch final : public RequestBase {
   virtual void Process() {
     // prefetch process...
     ::grpc::ByteBuffer reply;
-    // TODO(Yancey1989): execute the Block which containers prefetch ops
 
-    VLOG(3) << "RequestPrefetch Process in";
+    executor_->Run(*program_, scope_, blkid_, false, false);
```

Comment:
`Process` runs in a separated thread, `executor_` may be accessed in different threads at the same time, don't know whether this is safe.

Reply:
The problem is that the prefetch and optimize may happen at the same time, they will both access the lookup_table parameter. 

I think the final solution may be that table optimization also be a separate thread, and the prefetch thread and update thread try to get the same lock.

Currently, we will run update operators withing optimize block, should find a way to a avoid the conflict.

Reply:
> The problem is that the prefetch and optimize may happen at the same time

May not, for the current process, `prefetch request` would happen before sending gradients. And there is a **SEND BARRIER** to make sure that optimize process would happen after `prefetch request`

Reply:
But I think it's not threaded safe for the current implementation, because we use one scope to create the output variable, if there are more then two `prefetch request`, the output variable would be replaced, and the `serialize` function would be failed.
Maybe a way to solve this is to use different scope to create output var.

Reply:
We can create a sub_scope here to store the output variable.

Reply:
Agree with @jacquesqiao, but looks `NewScope` may also not threaded safe...
Maybe another way is to create multiple output vars with the different suffix such as `out_trainer0, out_trainer1` in `Distributed transpiler`.

Reply:
After discuss with @Yancey1989, we decided to use NewScope to run each Process.

Reply:
Done.